runtime/parachains/src/inclusion.rs

```diff
@@ -399,14 +402,20 @@ impl<T: Trait> Module<T> {
 					Error::<T>::CandidateNotInParentContext,
 				);
 
-				let code_upgrade_allowed = <paras::Module<T>>::last_code_upgrade(para_id, true)
-					.map_or(
-						true,
-						|last| last <= relay_parent_number &&
-							relay_parent_number.saturating_sub(last) >= config.validation_upgrade_frequency,
-					);
+				// if any, the code upgrade attempt is allowed.
+				let valid_upgrade_attempt = candidate.candidate.commitments.new_validation_code.is_none();
+					|| <paras::Module<T>>::last_code_upgrade(para_id, true)
+						.map_or(
+							true,
+							|last| last <= relay_parent_number &&
+								relay_parent_number.saturating_sub(last)
+									>= config.validation_upgrade_frequency,
+						);
```

Comment:
Maybe I'm misunderstanding, but it looks like 407-413 are an unexecuted bare closure not assigned to an identifier. Why is this there?

Reply:
because I accidentally added a semicolon on the previous line and this is supposed to be a boolean OR! wow...surprised rustc doesn't warn on that one.